### PR TITLE
[eslint] Fix core js-doc warning

### LINKED ISSUE #6750: [eslint] Fix warnings
When using eslint with the defined rules to validate the js scripts, many require-jsdoc and max-len warnings are fired. When an error breaks the compilation, it becomes tricky to spot the cause.
I volunteer to gradually fix all those warnings.

### PR DESCRIPTION
**Pull request by laemtl**
This PR:
* resolves all eslint js-doc warnings for the core files
* raises js-doc severity to error for the core files

### Step 1
```
rm -R nodes_modules/
npm install
```

### Step 2
```
make checkstatic
```
No warnings `js-doc` should be printed.

### Step 3
```
git checkout 'one-modiifed file'
make checkstatic
```
An error `js-doc` should be printed.

* resolves #6750 (partially)

### CODE REVIEW

File: htdocs/js/instrument_controlpanel_swaldeletedata.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
+ * Instrument seletion confirmation message
```
Comment (maltheism) [inline htdocs/js/instrument_controlpanel_swaldeletedata.js:2]: @laemtl spelling mistake for selection

File: htdocs/js/instrument_form_control.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
+/**
+ * Display a warming if a form element is not answered
```
Comment (maltheism) [inline htdocs/js/instrument_form_control.js:2]: spelling mistake for warning

File: jsx/Card.js [hunk 3]
```diff
@@ -37,9 +36,9 @@ class Card extends Component {
   }
 
   /**
-   * Render the React component
+   * Renders the React component.
    *
-   * @return {object}
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/Card.js:41]: `@return {react}` might be better here instead of string?

File: jsx/DataTable.js [hunk 4]
```diff
@@ -359,6 +408,11 @@ class DataTable extends Component {
     }
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/DataTable.js:414]: @return {react} might be better

File: jsx/DynamicDataTable.js [hunk 5]
```diff
@@ -63,13 +95,25 @@ class DynamicDataTable extends Component {
...
     console.error(errorCode + ': ' + errorMsg);
     this.setState({
       error: 'Error loading data',
     });
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/DynamicDataTable.js:115]: @return {react} might be better

File: jsx/Filter.js [hunk 6]
```diff
@@ -120,6 +132,11 @@ class Filter extends Component {
     }, []);
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/Filter.js:138]: @return {react} might be better

File: jsx/FilterForm.js [hunk 7]
```diff
@@ -170,6 +177,11 @@ class FilterForm extends Component {
     this.props.onUpdate(filter);
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/FilterForm.js:183]: @return {react} might be better

File: jsx/FilterableDataTable.js [hunk 8]
```diff
@@ -44,6 +48,11 @@ class FilterableDataTable extends Component {
     history.replaceState({}, '', '?');
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/FilterableDataTable.js:54]: @return {react} might be better

File: jsx/Form.js [hunk 9]
```diff
@@ -85,6 +97,11 @@ class FormElement extends Component {
     }
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/Form.js:103]: @return {react} might be better

File: jsx/Form.js [hunk 10]
```diff
@@ -185,6 +210,11 @@ class FieldsetElement extends Component {
     return formElementsHTML;
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/Form.js:216]: @return {react} might be better

File: jsx/Form.js [hunk 11]
```diff
@@ -266,6 +321,11 @@ class SearchableDropdown extends Component {
     }
   }
 
+  /**
+   * Renders the React component.
+   *
+   * @return {string} - HTML markup for the component
```
Comment (maltheism) [inline jsx/Form.js:327]: @return {react} might be better

**Comment by laemtl** [comment]
I usually take extra care to remove those files from any commit, might have missed it that time.

File: jsx/Card.js (hunk 3 above)
Comment (laemtl) [reply jsx/Card.js:41]: I thought about that too. I'm fine with either your suggestion or @return {JSX}.

Comment (maltheism) [reply jsx/Card.js:41]: I didn't think about the other two possibilities. Whatever you think is best will be great!

**Pull request merged**